payments_service/braintree/forms.py

```diff
@@ -82,3 +82,49 @@ def clean_pay_method_uri(self):
                 'paymethod by URI does not exist or belongs to another user'
             )
         return uri
+
+
+class SaleForm(forms.Form):
+    """
+    Input data for subscribing a user to a Braintree plan.
+    """
+    # A single use token representing the buyer's submitted payment method.
+    nonce = forms.CharField(max_length=255, required=False)
+    # Solitude URI to an existing payment method for this buyer.
+    paymethod = forms.CharField(max_length=255, required=False)
+    # ID from payments-config.
+    product_id = forms.CharField(max_length=255)
+    # A variable amount to pay for the product. This only applies to products
+    # like donations.
+    amount = forms.DecimalField(required=False)
+
+    def __init__(self, user, *args, **kwargs):
+        # This is the currently signed in user. It may be None.
+        self.user = user
+        super(SaleForm, self).__init__(*args, **kwargs)
+
+    def clean(self):
+        cleaned_data = super(SaleForm, self).clean()
+        if not cleaned_data.get('paymethod') and not cleaned_data.get('nonce'):
+            raise forms.ValidationError(
+                'either a payment nonce or paymethod URI is required')
```

Comment:
I had thought it was a good idea because I was relying on there being a nonce or a paymethod in order to decide whether to save the nonce for later. I had to remove that code because it wasn't working though. I'll remove this now too.